File: DuDoong-Api/src/main/java/band/gosrock/api/order/service/ConfirmOrderUseCase.java
package band.gosrock.api.order.service;


import band.gosrock.api.config.security.SecurityUtils;
import band.gosrock.api.order.mapper.OrderMapper;
import band.gosrock.api.order.model.dto.request.ConfirmOrderRequest;
import band.gosrock.api.order.model.dto.response.OrderResponse;
import band.gosrock.common.annotation.UseCase;
import band.gosrock.domain.domains.order.service.OrderConfirmService;
import band.gosrock.infrastructure.outer.api.tossPayments.dto.request.ConfirmPaymentsRequest;
import lombok.RequiredArgsConstructor;

@UseCase
@RequiredArgsConstructor
public class ConfirmOrderUseCase {

    private final OrderConfirmService orderConfirmService;

    private final OrderMapper orderMapper;

    public OrderResponse execute(String orderUuid, ConfirmOrderRequest confirmOrderRequest) {
        Long currentUserId = SecurityUtils.getCurrentUserId();
        ConfirmPaymentsRequest confirmPaymentsRequest =
                ConfirmPaymentsRequest.builder()
                        .paymentKey(confirmOrderRequest.getPaymentKey())
                        .amount(confirmOrderRequest.getAmount())
                        .orderId(orderUuid)
                        .build();
        String confirmOrderUuid =
                orderConfirmService.execute(confirmPaymentsRequest, currentUserId);
        return orderMapper.toOrderResponse(confirmOrderUuid);
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/order/service/ConfirmOrderUseCase.kt
package band.gosrock.api.order.service

import band.gosrock.api.order.model.dto.request.ConfirmOrderRequest
import band.gosrock.api.order.model.dto.response.OrderResponse
import band.gosrock.api.order.model.mapper.OrderMapper
import band.gosrock.common.annotation.UseCase
import band.gosrock.domain.domains.order.service.OrderConfirmService
import band.gosrock.infrastructure.outer.api.tossPayments.dto.request.ConfirmPaymentsRequest
import org.slf4j.LoggerFactory

@UseCase
class ConfirmOrderUseCase(
    private val orderConfirmService: OrderConfirmService,
    private val orderMapper: OrderMapper,
) {
    private val log = LoggerFactory.getLogger(ConfirmOrderUseCase::class.java)

    fun execute(userId: Long, orderUuid: String, confirmOrderRequest: ConfirmOrderRequest): OrderResponse {
        log.info("[ConfirmOrderUseCase][execute] 결제 확인 userId={} orderUuid={} amount={}", userId, orderUuid, confirmOrderRequest.amount)
        val confirmPaymentsRequest = ConfirmPaymentsRequest(
            paymentKey = confirmOrderRequest.paymentKey,
            amount = confirmOrderRequest.amount,
            orderId = orderUuid,
        )
        val confirmOrderUuid = orderConfirmService.execute(confirmPaymentsRequest, userId)
        return orderMapper.toOrderResponse(confirmOrderUuid)
    }
}
